src/main/java/net/rptools/maptool/client/ui/macrobuttons/buttongroups/ButtonGroup.java

```diff
@@ -121,18 +121,31 @@ public void drop(DropTargetDropEvent event) {
               data.toolTip,
               data.displayHotKey);
 
+      // if a reference is needed when moving instead of copying
+      MacroButtonProperties oldMacroProps = new MacroButtonProperties(tempProperties);
+
       if (panelClass.equals("GlobalPanel")) {
         event.acceptDrop(event.getDropAction());
         tempProperties.setGroup(
             getMacroGroup()); // assign the group you are dropping it into, rather than the original
         if (!tempProperties.isDuplicateMacro("GlobalPanel", null)) {
+          if (event.getDropAction() == 1) { // DnDConstants says 1 is copy,
+            // but this is for reversed behavior (copy as default)
+            getArea().getPropertiesList().remove(oldMacroProps);
+            getArea().repaint();
+          }
```

Comment:
Suggestion:
- extract this piece of code into a small private method which can be re-used instead of repeating the same behavior
- use a constant for 1 (as you are adding a comment, make a constant instead)

Reply:
Don't know why I didn't do that first one, thanks! As for using the provided constants, I originally chose not to because it has counter-intuitive meaning (ACTION_COPY = 1, but the if statement is meant to be "if the token was moved"). This is a thing because when createDefaultDragGestureRecognizer is set to recognize ACTION_COPY_OR_MOVE in MacroButton, the default drag action becomes moving, with a dropAction code of 2. Holding Ctrl while dragging gives a dropAction code of 1. I'm not sure why this is, but I'm also quite unfamiliar with Swing, so it could be something obvious that I'm not aware of!

Reply:
I am not very much familiar with Swing as well, so take my advice with caution :) 

Looking at DropTargetDropEvent.java, it looks like pressing "Shift" (rather thank Ctrl) should generate an ACTION_MOVE event, which would be a little better. I think you could get rid of the comment at that point.

Reply:
Right now, the GestureRecognizer in MacroButton interprets a non-modified drag (no keys pressed) as ACTION_MOVE, and shift appears to do nothing. I'm really not sure why, and it changed after I had the GestureRecognizer recognize ACTION_MOVE_OR_COPY instead of ACTION_COPY. I'll dig around in a bit and see if I can maybe get it to behave as the documentation says it should (I can't get an ACTION_LINK event while dragging, for instance). In the meanwhile, it is (externally) working as intended.